app/src/extra/java/org/wikipedia/donate/GooglePayViewModel.kt

```diff
@@ -85,24 +84,7 @@ class GooglePayViewModel : ViewModel() {
                 disclaimerMonthlyCancel = response.query?.allmessages?.find { it.name == MSG_DISCLAIMER_MONTHLY_CANCEL }?.content?.replace("$1", WikipediaApp.instance.getString(R.string.donate_email))
             }
 
-            // The paymentMethods API is rate limited, so we cache it manually.
-            val now = Instant.now().epochSecond
-            if (abs(now - Prefs.paymentMethodsLastQueryTime) > TimeUnit.DAYS.toSeconds(7)) {
-                Prefs.paymentMethodsMerchantId = ""
-                Prefs.paymentMethodsGatewayId = ""
-
-                val paymentMethodsCall = async {
-                    ServiceFactory.get(WikiSite(GooglePayComponent.PAYMENTS_API_URL))
-                        .getPaymentMethods(currentCountryCode)
-                }
-                paymentMethodsCall.await().response?.let { response ->
-                    Prefs.paymentMethodsLastQueryTime = now
-                    response.paymentMethods.find { it.type == GooglePayComponent.PAYMENT_METHOD_NAME }?.let {
-                        Prefs.paymentMethodsMerchantId = it.configuration?.merchantId.orEmpty()
-                        Prefs.paymentMethodsGatewayId = it.configuration?.gatewayMerchantId.orEmpty()
-                    }
-                }
-            }
+            async { updatePaymentMethodsPreferences() }.await()
```

Comment:
This doesn't need to be inside `async{}.await()`

Reply:
Sure!